examples/dbscan/ArborX_DetailsDBSCANVerification.hpp

```diff
@@ -123,14 +123,18 @@ bool verifyBoundaryPointsConnectToCorePoints(ExecutionSpace const &exec_space,
 // Check that cluster indices are unique
 template <typename ExecutionSpace, typename IndicesView, typename OffsetView,
           typename ClusterView>
-bool verifyClustersAreUnique(ExecutionSpace const &, IndicesView indices,
-                             OffsetView offset, ClusterView clusters,
-                             int core_min_size)
+bool verifyClustersAreUnique(ExecutionSpace const &exec_space,
+                             IndicesView indices, OffsetView offset,
+                             ClusterView clusters, int core_min_size)
 {
   int n = clusters.size();
 
-  auto clusters_host =
-      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, clusters);
+  decltype(Kokkos::create_mirror_view(Kokkos::HostSpace{},
+                                      std::declval<ClusterView>()))
+      clusters_host(Kokkos::ViewAllocateWithoutInitializing(
+                        "ArborX::DBSCAN::clusters_host"),
+                    clusters.size());
+  Kokkos::deep_copy(exec_space, clusters_host, clusters);
```

Comment:
Why did you change this??

Reply:
Because `clusters` were being modified in this verification check. Thus, if you run `verify` and then do postprocessing, it was screwed up. This has been a long unnoticed bug.

Reply:
So you need a "create mirror and copy" (instead of "create a mirror **view** and copy")?
Go ahead and create a function helper for that.  I don't think Kokkos has anything.

Reply:
Added a comment stating what we want to do here. I think it's too early to introduce this helper routine until we have at least another case for it.